File: static/coin_config.js
/* English-only mining metadata. These values intentionally bypass i18n. */
window.COIN_CONFIG = {
    BTC: {
        ticker: 'BTC', key: 'btc', name: 'Bitcoin', icon: '/static/images/coins/btc.svg', algorithm: 'SHA-256',
        hash_unit: ['TH/s', 'GH/s', 'PH/s'], network_hash_unit: ['EH/s', 'PH/s', 'TH/s'],
    },
    LTC: {
        ticker: 'LTC', key: 'ltc', name: 'Litecoin', icon: '/static/images/coins/ltc.svg', algorithm: 'Scrypt',
        hash_unit: ['MH/s', 'GH/s'], network_hash_unit: ['MH/s', 'GH/s', 'TH/s'],
    },
    DOGE: {
        ticker: 'DOGE', key: 'doge', name: 'Dogecoin', icon: '/static/images/coins/doge.svg', algorithm: 'Scrypt',
        hash_unit: ['MH/s', 'GH/s'], network_hash_unit: ['MH/s', 'GH/s', 'TH/s'],
    },
    KAS: {
        ticker: 'KAS', key: 'kas', name: 'Kaspa', icon: '/static/images/coins/kas.svg', algorithm: 'kHeavyHash',
        hash_unit: ['GH/s', 'TH/s', 'PH/s'], network_hash_unit: ['TH/s', 'PH/s'],
    },
    ETC: {
        ticker: 'ETC', key: 'etc', name: 'Ethereum Classic', icon: '/static/images/coins/etc.svg', algorithm: 'Ethash',
        hash_unit: ['MH/s', 'GH/s'], network_hash_unit: ['MH/s', 'GH/s', 'TH/s'],
    },
    ZEC: {
        ticker: 'ZEC', key: 'zec', name: 'Zcash', icon: '/static/images/coins/zec.svg', algorithm: 'Equihash',
        hash_unit: ['Sol/s', 'kSol/s', 'MSol/s', 'GSol/s'], network_hash_unit: ['Sol/s', 'kSol/s', 'MSol/s', 'GSol/s'],
    },
    DASH: {
        ticker: 'DASH', key: 'dash', name: 'Dash', icon: '/static/images/coins/dash.svg', algorithm: 'X11',
        hash_unit: ['GH/s', 'TH/s', 'PH/s'], network_hash_unit: ['GH/s', 'TH/s', 'PH/s'],
    },
    XMR: {
        ticker: 'XMR', key: 'xmr', name: 'Monero', icon: '/static/images/coins/xmr.svg', algorithm: 'RandomX',
        hash_unit: ['H/s', 'kH/s', 'MH/s', 'GH/s'], network_hash_unit: ['H/s', 'kH/s', 'MH/s', 'GH/s'],
    },
    RVN: {
        ticker: 'RVN', key: 'rvn', name: 'Ravencoin', icon: '/static/images/coins/rvn.svg', algorithm: 'KawPow',
        hash_unit: ['MH/s', 'GH/s', 'TH/s'], network_hash_unit: ['MH/s', 'GH/s', 'TH/s'],
    },
    ERG: {
        ticker: 'ERG', key: 'erg', name: 'Ergo', icon: '/static/images/coins/erg.svg', algorithm: 'Autolykos',
        hash_unit: ['MH/s', 'GH/s', 'TH/s'], network_hash_unit: ['MH/s', 'GH/s', 'TH/s'],
    },
    ALPH: {
        ticker: 'ALPH', key: 'alph', name: 'ALEPHIUM', icon: '', algorithm: 'Blake3',
        hash_unit: ['GH/s', 'TH/s', 'PH/s'], network_hash_unit: ['GH/s', 'TH/s', 'PH/s'],
    },
    IRONFISH: {
        ticker: 'IRONFISH', key: 'ironfish', name: 'IRONFISH', icon: '', algorithm: 'FishHash',
        hash_unit: ['MH/s', 'GH/s', 'TH/s'], network_hash_unit: ['MH/s', 'GH/s', 'TH/s'],
    },
};

window.CoinConfig = window.COIN_CONFIG;

File: market_data/coin_config.py
"""External data-source metadata, separate from ROI calculation settings."""


WHAT_TO_MINE_URL = "https://whattomine.com/coins.json"


MARKET_COIN_CONFIG = {
    "BTC": {
        "coingecko_id": "bitcoin",
        "recommended_network_unit": "EH/s",
        "average_block_time": 600,
        "average_block_reward": 3.125,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [
            {"provider": "mempool", "url": "https://mempool.space/api/v1/mining/hashrate/1m"},
        ],
    },
    "LTC": {
        "coingecko_id": "litecoin",
        "recommended_network_unit": "TH/s",
        "average_block_time": 150,
        "average_block_reward": 6.25,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [{"provider": "blockchair", "url": "https://api.blockchair.com/litecoin/stats"}],
    },
    "DOGE": {
        "coingecko_id": "dogecoin",
        "recommended_network_unit": "TH/s",
        "average_block_time": 60,
        "average_block_reward": 10000,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [{"provider": "blockchair", "url": "https://api.blockchair.com/dogecoin/stats"}],
    },
    "KAS": {
        "coingecko_id": "kaspa",
        "recommended_network_unit": "PH/s",
        "hashrate_providers": [{"provider": "kaspa", "url": "https://api.kaspa.org/info/hashrate"}],
    },
    "ETC": {
        "coingecko_id": "ethereum-classic",
        "recommended_network_unit": "TH/s",
        "average_block_time": 13.3333333333,
        "average_block_reward": 1.5,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [
            {"provider": "2miners", "url": "https://etc.2miners.com/api/stats"},
            {"provider": "whattomine", "url": WHAT_TO_MINE_URL, "coin_id": "EthereumClassic", "coin_tag": "ETC"},
        ],
    },
    "ZEC": {
        "coingecko_id": "zcash",
        "recommended_network_unit": "GSol/s",
        "average_block_time": 75,
        "average_block_reward": 1.5625,
        "mining_reward_ratio": 0.8,
        "hashrate_providers": [
            {"provider": "blockchair", "url": "https://api.blockchair.com/zcash/stats"},
            {"provider": "2miners", "url": "https://zec.2miners.com/api/stats"},
        ],
    },
    "DASH": {
        "coingecko_id": "dash",
        "recommended_network_unit": "TH/s",
        "hashrate_providers": [{"provider": "blockchair", "url": "https://api.blockchair.com/dash/stats"}],
    },
    "XMR": {
        "coingecko_id": "monero",
        "recommended_network_unit": "GH/s",
        "average_block_time": 120,
        "average_block_reward": 0.6,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [{"provider": "xmrchain", "url": "https://xmrchain.net/api/networkinfo"}],
    },
    "RVN": {
        "coingecko_id": "ravencoin",
        "recommended_network_unit": "TH/s",
        "average_block_time": 60,
        "average_block_reward": 1250,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [
            {"provider": "2miners", "url": "https://rvn.2miners.com/api/stats"},
            {"provider": "whattomine", "url": WHAT_TO_MINE_URL, "coin_id": "Ravencoin", "coin_tag": "RVN"},
        ],
    },
    "ERG": {
        "coingecko_id": "ergo",
        "recommended_network_unit": "TH/s",
        "average_block_time": 120,
        "average_block_reward": 3,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [
            {"provider": "ergo", "url": "https://api.ergoplatform.com/info"},
            {"provider": "whattomine", "url": WHAT_TO_MINE_URL, "coin_id": "Ergo", "coin_tag": "ERG"},
        ],
    },
    "ALPH": {
        "coingecko_id": "alephium",
        "recommended_network_unit": "PH/s",
        "hashrate_providers": [{"provider": "alephium", "url": "https://backend.mainnet.alephium.org/charts/hashrates"}],
    },
    "IRONFISH": {
        "coingecko_id": "iron-fish",
        "recommended_network_unit": "TH/s",
        "average_block_time": 60,
        "average_block_reward": 17.25,
        "mining_reward_ratio": 1.0,
        "hashrate_providers": [
            {"provider": "whattomine", "url": WHAT_TO_MINE_URL, "coin_id": "IronFish", "coin_tag": "IRON"},
        ],
    },
}


def get_market_config(coin):
    """Return source metadata for a ticker, or None for unsupported input."""
    return MARKET_COIN_CONFIG.get(str(coin or "").strip().upper())
